File: 3653_XOR_After_Range_Multiplication_Queries_I/solution.c
#define M 1000000007
int xorAfterQueries(int* a, int n, int** qs, int qn, int* qc) {
    for (int i = 0; i < qn; i++) {
        int l = qs[i][0], r = qs[i][1], k = qs[i][2], v = qs[i][3];
        for (int j = l; j <= r; j += k) {
            long long x = a[j];
            x = (x * v) % M;
            a[j] = (int)x;
        }
    }
    int x = 0;
    for (int i = 0; i < n; i++) x ^= a[i];
    return x;
}

File: 3653_XOR_After_Range_Multiplication_Queries_I/solution.py
class Solution:
    def xorAfterQueries(self, nums: List[int], queries: List[List[int]]) -> int:
        m = 10**9 + 7
        for q in queries:
            l, r, k, v = q
            i = l
            while i <= r:
                nums[i] = (nums[i] * v) % m
                i += k
        x = 0
        for n in nums: x ^= n
        return x
